### PR TITLE
Various doxygen documentation fixes

### PR DESCRIPTION
**Pull request by Maetveis**
cc @saadrahim

### CODE REVIEW

File: include/hip/hip_runtime_api.h [hunk 1]
```diff
@@ -235,7 +231,7 @@ typedef enum __HIP_NODISCARD hipError_t {
     hipErrorPriorLaunchFailure = 53,
     hipErrorInvalidDeviceFunction = 98,
     hipErrorNoDevice = 100,  ///< Call to hipGetDeviceCount returned 0 devices
-    hipErrorInvalidDevice = 101,  ///< DeviceID must be in range 0...#compute-devices.
+    hipErrorInvalidDevice = 101,  ///< DeviceID must be in range 0...\#compute-devices.
```
Comment (gargrahul) [inline include/hip/hip_runtime_api.h:234]: This change is breaking the CI build.

File: include/hip/hip_runtime_api.h [hunk 2]
```diff
@@ -696,7 +691,7 @@ enum hipLimit_t {
 #define hipCooperativeLaunchMultiDeviceNoPreSync 0x01
 #define hipCooperativeLaunchMultiDeviceNoPostSync 0x02
 #define hipCpuDeviceId ((int)-1)
-#define hipInvalidDeviceId ((int)-2)
+#define hipErrorInvalidDeviceId ((int)-2)
```
Comment (gargrahul) [inline include/hip/hip_runtime_api.h]: This change is breaking CI build.

File: include/hip/hip_runtime_api.h (hunk 1 above)
Comment (gargrahul) [inline include/hip/hip_runtime_api.h:243]: This change is breaking the CI build.

File: include/hip/hip_runtime_api.h (hunk 2 above)
Comment (Maetveis) [reply include/hip/hip_runtime_api.h]: Yup, I did not check hipamd, so I was confused on how this is used, turns out these two are separate things:
- `hipErrorInvalidDevice` an error that is returned when the device id supplied to an api call is not valid
- `hipInvalidDeviceId` is a possible return value for `hipMemRangeAttributePreferredLocation`, ironically this is not documented anywhere.

Should be fixed now.

File: include/hip/hip_runtime_api.h [hunk 3]
```diff
@@ -202,14 +200,12 @@ typedef struct hipPointerAttribute_t {
 #define __HIP_NODISCARD
 #endif
 
+// Developer note - when updating these, update the hipErrorName and hipErrorString functions in
+// NVCC and HCC paths Also update the hipCUDAErrorTohipError function in NVCC path.
```
Comment (gargrahul) [inline include/hip/hip_runtime_api.h]: HCC should not be used.

**Comment by gargrahul** [comment]
@Maetveis Please rebase this change on latest develop branch.

**Review by chrispaquot — DISMISSED** [review]

**Review by saadrahim — DISMISSED** [review]
@gargrahul can you help merge this?

**Pull request closed without merging**